storey/table.py

```diff
@@ -244,6 +288,151 @@ def __getitem__(self, key):
+                    if done_job.callback:
+                        await done_job.callback(done_job.extra_data, completed)
+                self._pending_by_key[job_key].in_flight = []
+
+                # If we got more pending events for the same key process them
+                if self._pending_by_key[job_key].pending:
+                    self._pending_by_key[job_key].in_flight = self._pending_by_key[job_key].pending
+                    self._pending_by_key[job_key].pending = []
+
+                    future_task = self._safe_process_events(self._pending_by_key[job_key].in_flight)
+                    tail_position = received_job_count + self._q.qsize()
+                    jobs_at_tail = self_sent_jobs.get(tail_position, [])
+                    jobs_at_tail.append((job, asyncio.get_running_loop().create_task(future_task)))
+                    self_sent_jobs[tail_position] = jobs_at_tail
+                else:
+                    del self._pending_by_key[job_key]
+        except BaseException as ex:
+            if task and task is not _termination_obj:
+                self._changed_keys.discard(task[0].key)
+                if task[0].extra_data and task[0].extra_data._awaitable_result:
+                    task[0].extra_data._awaitable_result._set_error(ex)
+            if not self._q.empty():
+                await self._q.get()
+            raise ex
+
+    async def _terminate(self):
+        if self._flush_exception is not None:
+            raise self._flush_exception
+        if not self._terminated:
+            self._terminated = True
+            for key in self._changed_keys:
+                await self._persist(_PersistJob(key, None, None), from_terminate=True)
+
+            await self._q.put(_termination_obj)
+            await self._worker_awaitable
+            if self._flush_task:
+                self._flush_task.cancel()
+                with suppress(asyncio.CancelledError):
+                    await self._flush_task
+            self._q = None
+            self._flush_task = None
+            self._flush_exception = None
+
+    async def _persist(self, job, from_terminate=False):
+        if self._flush_exception is not None:
+            raise self._flush_exception
+        if not self._q:
+            self._q = asyncio.queues.Queue(self._max_updates_in_flight)
+            self._worker_awaitable = asyncio.get_running_loop().create_task(self._persist_worker())
+            if not from_terminate:
+                # for flow reuse
+                self._terminated = False
+
+        if self._worker_awaitable.done():
+            await self._worker_awaitable
+            raise FlowError("Persist worker has already terminated")
+        else:
+            # Initializing the key with 2 lists. One for pending requests and one for requests that an update request has been issued for.
+            if isinstance(job.key, list):
+                # list can't be key in a dictionary
+                job_key = str(job.key)
```

Comment:
Should probably be the hashing function.

Reply:
this is the code in development 
https://github.com/mlrun/storey/blob/122fd04da3301c09f0aad058682a9d19041cf67d/storey/flow.py#L735

Reply:
So bug in dev?

Reply:
don't know. i'll wait for katya to take a look and get back to us